lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
The reason for adding xl_r is because the Fortran compiler.  The xlf*_r compilers link to the threaded Fortran runtime compared to the xlf* (non _r) compilers.  Sometimes, it is necessary to build w/the the threaded version.  For the C compiler, not much of a different between the the _r and non _r configurations.